ui/app.py

```diff
@@ -58,22 +62,29 @@ def tldr(text_input):
             "Which tl;dr best fits ?",
             tldr_text_list)
         if tldr_text:  
-            return tldr_text
-    else:
-        return None
+            dalle2(tldr_text)
 
 
 def dalle2(text_input):
-    if text_input:
-        img_response = openai.Image.create(prompt=text_input, n=1, size="1024x1024")
-        image_url = img_response.data[0].url
-        st.image(image_url)
+    st.subheader("Your DALL-E representation of the takeaways")
+    if st.button("DALLE") and text_input:
+        img_response = openai.Image.create(prompt=text_input, n=3, size='256x256')
+        image_urls = [img_response.data[idx].url for idx in range(len(img_response))]
+        st.image(image_urls)
     else:
         st.image(DEFAULT_IMG_URL, width=500)
 
 
 if __name__ == "__main__":
     init()
-    text_input = input_initial_data()
-    text = tldr(text_input)
-    dalle2(text)
+    question = get_search_question()
+    # TODO: semantic search using question and get list of concerns
+    results = ["batteries", "aluminium usage", "electricity demand", "..."]
+    return_search_results(results)
+    # TODO: search results to suggested action
```

Comment:
and this one!

Reply:
maybe for @lfhvn ? :) 
searching policies for suggesting some actions via https://github.com/climatepolicyradar/open-data